src/lib.rs

```diff
@@ -1234,9 +1234,14 @@ impl From<Attributes> for PutOptions {
     }
 }
 
+// See <https://github.com/apache/arrow-rs-object-store/issues/339>.
+#[doc(hidden)]
+#[deprecated(note = "Use PutMultipartOptions")]
```

Comment:
Can we also please add an `since` annotation so we know in the future when to remove this? 

For example like this
https://github.com/apache/arrow-rs/blob/639b5bb93e5a152a437b93a25ead8095a3866a9b/parquet/src/file/footer.rs#L47

I believe it will be released in `0.13.0`:
- https://github.com/apache/arrow-rs-object-store/issues/367

Reply:
done